lib/Dialect/HW/HWOps.cpp

```diff
@@ -1710,6 +1710,14 @@ LogicalResult ArrayCreateOp::verify() {
   return success();
 }
 
+Value ArrayCreateOp::getUniformElement() {
+  auto first = getInputs()[0];
```

Comment:
It kind of looks like we can create 0-length arrays with this operation, I don't see a verifier that the `number of operands > 0`

Reply:
Ok I think its intended that arrays can't be 0 length, so I wouldn't worry about this.

Reply:
It probably still can't hurt to be more cautious here :grin:

Reply:
IIRC we don't allow zero-length arrays since we don't support `i0`, which we may in the future. So zero-length arrays may be subject to reconsideration. It would eliminate a number of corner cases.

Reply:
:+1: I've added check so this function won't be one of the things we'll have to change once we decide to allow zero-length arrays.